google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java

```diff
@@ -537,8 +615,37 @@ public Builder setMaxAckExtensionPeriod(Duration maxAckExtensionPeriod) {
      * <p>MaxDurationPerAckExtension configuration can be disabled by specifying a zero duration.
      */
     public Builder setMaxDurationPerAckExtension(Duration maxDurationPerAckExtension) {
-      Preconditions.checkArgument(maxDurationPerAckExtension.toMillis() >= 0);
+      // If a non-default min is set, make sure min is less than max
+      Preconditions.checkArgument(
+          maxDurationPerAckExtension.toMillis() >= 0
+              && (this.minDurationPerAckExtensionDefaultUsed
+                  || (this.minDurationPerAckExtension.toMillis()
+                      < maxDurationPerAckExtension.toMillis())));
       this.maxDurationPerAckExtension = maxDurationPerAckExtension;
+      this.maxDurationPerAckExtensionDefaultUsed = false;
+      return this;
+    }
+
+    /**
+     * Set the lower bound for a single mod ack extention period.
+     *
+     * <p>The ack deadline will continue to be extended by up to this duration until
+     * MinAckExtensionPeriod is reached. Setting MinDurationPerAckExtension bounds the minimum
+     * amount of time before a mesage re-delivery in the event the Subscriber fails to extend the
+     * deadline.
+     *
+     * <p>MinDurationPerAckExtension configuration can be disabled by specifying a zero duration.
+     */
+    public Builder setMinDurationPerAckExtension(Duration minDurationPerAckExtension) {
+      // If a non-default max is set, make sure min is less than max
+      Preconditions.checkArgument(
+          maxDurationPerAckExtension.toMillis() >= 0
```

Comment:
Should we be checking that the minDurationPerAckExtension > 0 (not max)

Reply:
yes, good catch